src/main/java/org/sonarlint/intellij/ui/tree/SecurityHotspotTreeModelBuilder.java

```diff
@@ -213,6 +217,57 @@ private void setRootSecurityHotspots(VirtualFile file, Iterable<LiveSecurityHots
     }
   }
 
+  private void copyToFilteredNodes() {
+    nonFilteredNodes.clear();
+    Collections.list(summary.children()).forEach(e -> {
+      var securityHotspotNode = (LiveSecurityHotspotNode) e;
+      nonFilteredNodes.add((LiveSecurityHotspotNode) securityHotspotNode.clone());
+    });
+  }
+
+  public void showLocalSecurityHotspots() {
+    Collections.list(summary.children()).forEach(e -> model.removeNodeFromParent((LiveSecurityHotspotNode) e));
+
+    for (var securityHotspotNode : nonFilteredNodes) {
+      if (securityHotspotNode.getHotspot().getServerFindingKey() == null) {
+        var idx = summary.insertLiveSecurityHotspotNode(securityHotspotNode, new LiveSecurityHotspotNodeComparator());
+        var newIdx = new int[]{idx};
+        model.nodesWereInserted(summary, newIdx);
+        model.nodeChanged(summary);
+      }
+    }
+
+    model.reload();
+  }
+
+  public void showSonarQubeSecurityHotspots() {
+    Collections.list(summary.children()).forEach(e -> model.removeNodeFromParent((LiveSecurityHotspotNode) e));
+
+    for (var securityHotspotNode : nonFilteredNodes) {
+      if (securityHotspotNode.getHotspot().getServerFindingKey() != null) {
+        var idx = summary.insertLiveSecurityHotspotNode(securityHotspotNode, new LiveSecurityHotspotNodeComparator());
+        var newIdx = new int[]{idx};
+        model.nodesWereInserted(summary, newIdx);
+        model.nodeChanged(summary);
+      }
+    }
+
+    model.reload();
+  }
+
+  public void showAllSecurityHotspots() {
+    Collections.list(summary.children()).forEach(e -> model.removeNodeFromParent((LiveSecurityHotspotNode) e));
+
+    for (var securityHotspotNode : nonFilteredNodes) {
+      var idx = summary.insertLiveSecurityHotspotNode(securityHotspotNode, new LiveSecurityHotspotNodeComparator());
+      var newIdx = new int[]{idx};
+      model.nodesWereInserted(summary, newIdx);
+      model.nodeChanged(summary);
+    }
+
+    model.reload();
+  }
+
```

Comment:
There is a bunch of duplication in those 3 show methods. What about using the `SecurityHotspotFilters` enum and introduce a `shouldInclude` method ? This way you just pass the filter to this class and you have a single show method

Reply:
You're right, way better :+1: